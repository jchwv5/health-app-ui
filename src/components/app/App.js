/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import {
  BrowserRouter, Route, Switch, Redirect
} from 'react-router-dom';
import PatientsTable from '../patients/PatientsTable';
import PatientDetails from '../patient-details/PatientDetails';

/**
 * Main app for router
 * @name App
 * @returns component
 */
const App = () => (
  <div className="app-container">
    <BrowserRouter>
      <Switch>
        <div className="content-container">
          <Route
            exact
            path="/"
            render={() => (
              <Redirect to="/patients" />
            )}
          />
          <Route exact path="/patients" render={() => <PatientsTable />} />
          <Route exact path="/patients/details/:patientId" render={() => <PatientDetails />} />
        </div>
      </Switch>
    </BrowserRouter>
  </div>
);

export default App;
