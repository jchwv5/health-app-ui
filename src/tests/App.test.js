import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../components/app/App';

it('renders successfully', () => {
  const div = document.createElement('div');
  ReactDom.render(<App />, div);
  ReactDom.unmountComponentAtNode(div);
});

it('App component matches snapshot', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
