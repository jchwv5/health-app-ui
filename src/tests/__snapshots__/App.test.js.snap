// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="app-container"
>
  <div
    className="content-container"
    computedMatch={
      Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
    location={
      Object {
        "hash": "",
        "key": "9vb2mj",
        "pathname": "/patients",
        "search": "",
        "state": undefined,
      }
    }
  >
    <div>
      <h3
        className="pateintsHeader"
      >
        Patients
      </h3>
      <table
        className="patientsTable"
      >
        <thead>
          <tr
            className="noHover"
          >
            <th
              className="deleteHeader"
            />
            <th>
              First Name
            </th>
            <th>
              Last Name
            </th>
            <th>
              Age
            </th>
            <th>
              Gender
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <div
        className="wrapper"
      >
        <button
          className="addPatientButton"
          onClick={[Function]}
          type="submit"
        >
          Add a Patient
        </button>
      </div>
    </div>
  </div>
</div>
`;
